src/lib.rs

```diff
@@ -354,6 +355,7 @@ macro_rules! entry {
                 struct MyVisitor;
 
                 #[derive(Deserialize)]
+                #[serde(deny_unknown_fields)]
```

Comment:
You proposed to warn instead in the original issue. I'm aware that we have no infrastructure for warnings at the moment, however, I do agree with you that we should try to stay forward-compatible and that denying parsing is too strict here. Would you be okay with removing this from the PR for now?

Reply:
Do you want just the doc fixes in this PR and the warnings in another one or should I create warnings in this PR as well?

About warnings: I can certainly do that but I might not find the most elegant solutions and will most likely have to change the API in some parts.

Reply:
I would prefer separate PRs so we can land the docs changes sooner!

It would be great if you did. I think there are two possible designs: Returning a `Result<(T, NonCriticalDiags[]), CriticalDiags[]>` or returning a pointer to a tracer struct that contains the warnings (like Typst does: https://github.com/typst/typst/blob/c20b6ec6e157b753e90aa60ccd5b50f6f31cb7f6/crates/typst/src/lib.rs#L84).